src/Microsoft.Health.Fhir.Api/Features/ActionResults/OperationResult.cs

```diff
@@ -0,0 +1,80 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Net;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Fhir.Core.Features.Export;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Api.Features.ActionResults
+{
+    public class OperationResult : ActionResult
+    {
+        public OperationResult()
+        {
+        }
+
+        public OperationResult(OperationOutcome error)
+        {
+            EnsureArg.IsNotNull(error, nameof(error));
+
+            Error = error;
+        }
+
+        public OperationResult(ExportJobResult jobResult)
+        {
+            EnsureArg.IsNotNull(jobResult, nameof(jobResult));
+
+            JobResult = jobResult;
+        }
+
+        public HttpStatusCode? StatusCode { get; set; }
+
+        public OperationOutcome Error { get; }
```

Comment:
Could this just be passed in as a type parameter? e.g., OperationResult<OperationOutcome> or OperationResult<ExportJobResult>? If it's not being used in an array, it might be cleaner.